app/controllers/management/site_pages_controller.rb

```diff
@@ -125,6 +126,11 @@ def toggle_enable
   # Use callbacks to share common setup or constraints between actions.
   def set_site
     @site = Site.find_by({slug: params[:site_slug]})
+    routes = Route.where("site_id = ?", @site.id)
+
+    unless (routes.empty?)
+      @url = routes.first.host
```

Comment:
I didn't understand this part: you're checking if it has routes, but then you just use the first one?